### ISSUE TITLE
conflicting between SpeedAction and TrajectoryAction(TimeReference =none) at latest commit

### ISSUE THREAD

**Issue by bleach31** [esmini/esmini#157]
I found the problem of the title.

here is scenario file for testing this.
[trajectory_with_speedAction.zip](https://github.com/esmini/esmini/files/7166229/trajectory_with_speedAction.zip)

### testing scenario overview

1. `SpeedAction` with 3 m/s at init
1. `SpeedAction` in the Story-Act with 30 m/s, acceleration 3 m/s/s (at sim_time = 2), priority="overwrite"
1. `FollowTrajectoryAction` in the Story-Act with <TimeReference> <None/> (at sim_time = 5), priority="overwrite"

The expectation is that `SpeedAction` is longitudinal and `FollowTrajectoryAction` with no TimeReference is lateral, so they should not conflict. But when the trajectory is triggered, this cancels the speed action.
Therefore, the vehicle never reaches the target velocity.
The key lines in the esmini output are :

```
5.009: Event speed_Event ended, overwritten by event trajectory_Event
5.009: Stopping object Host speed_Action on conflicting longitudinal domain(s)
```

Here is the full output:
[log.txt](https://github.com/esmini/esmini/files/7166227/log.txt)

I think that the cause is hidden in f5b3b40c58cdc4d0b6691891f1c096feab9ea562, because it works fine in v2.13.4.

**Comment by eknabevcc** [comment]
Excellent finding again. Thanks for reporting!

When adding support for solving conflicting actions it was wrongly assumed that FollowTrajectoryAction always affects both domains. This is a bug, and it should now be fixed in commit 6942c96b103f56b52c54440c1583e8ecf6ef2049.

However, in your scenario the trajectory_Event is overwriting the speed_Event, not due to (assumed) conflicting domains, but due to the specified priority between the events in the maneuver. This priority apply to complete event regardless of domains of its actions.

So, to my understanding, you can easily fix it by changing priority from "overwrite" to "parallel".
Another option is to put the events in separate maneuvers. Attached zip contains both variants.
[two-fix-variants.zip](https://github.com/esmini/esmini/files/7166982/two-fix-variants.zip)

To clarify: Yes, you found a bug, and it should now be solved. But the scenario also needs minor modification to work as expected.

Please test (you can pull master) and report back if it works to your expectations. In which case I think we should release these fixes 👍

**Comment by bleach31** [comment]
Thank you very much for always responding sooo quickly.

As soon as I discovered the commit, I tested it.

> However, in your scenario the trajectory_Event is overwriting the speed_Event, not due to (assumed) conflicting domains, but due to the specified priority between the events in the maneuver. This priority apply to complete event regardless of domains of its actions.

Yes, that's what I was wondering now!

Thank you for your comment. It is written in the specification.

https://releases.asam.net/OpenSCENARIO/1.0.0/ASAM_OpenSCENARIO_BS-1-2_User-Guide_V1-0-0.html#_events
> overwrite | All other Events in the scope are stopped and the Event starts.

I close this issue.

**Comment by eknabevcc** [comment]
FYI: These fixes are included in release [v2.13.5](https://github.com/esmini/esmini/releases/tag/v2.13.5)
